### PR TITLE
Fixes #6285 - Settings API does not parse incoming values

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

**Comment by domcleal** [comment]
Would you mind adding functional test(s) for this?

File: app/controllers/api/v2/base_controller.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def metadata_page
       end
 
       def metadata_per_page
-        @per_page ||= params[:per_page].present? ? params[:per_page].to_i : Setting::General.entries_per_page
+        @per_page ||= params[:per_page].present? ? params[:per_page].to_i : Setting::General.entries_per_page.to_i
```
Comment (domcleal) [inline app/controllers/api/v2/base_controller.rb]: Why this change?

Comment (domcleal) [inline app/controllers/api/v2/base_controller.rb]: Ah ok, I see.. maybe add a migration to tidy up broken data?  Or otherwise it'd also need adding to lib/core_extensions.rb.

Comment (tstrachota) [inline app/controllers/api/v2/base_controller.rb]: The migration is a good idea. I'll add it.

**Comment by tstrachota** [comment]
I updated the PR with tests and the migration.

File: db/migrate/20140623144932_fix_integer_settings.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class FixIntegerSettings < ActiveRecord::Migration
+
+  def change
```
Comment (domcleal) [inline db/migrate/20140623144932_fix_integer_settings.rb]: nitpick, this should be in a self.up method and self.down should be empty

File: app/controllers/api/v2/base_controller.rb (hunk 1 above)
Comment (domcleal) [inline app/controllers/api/v2/base_controller.rb]: This can be removed now then

File: test/functional/api/v1/settings_controller_test.rb [hunk 3]
```diff
@@ -22,4 +22,11 @@ class Api::V1::SettingsControllerTest < ActionController::TestCase
     assert_response :success
   end
 
+  test "should parse string values" do
+    setting_id = Setting.where(:settings_type => 'integer').first.id
+    put :update, { :id => setting_id, :setting => { :value => "100" } }
+    assert_response :success
+    assert Setting.where(:settings_type => 'integer').first.value == 100
```
Comment (domcleal) [inline test/functional/api/v1/settings_controller_test.rb]: nitpick (ditto below), use `assert_equal 100, Setting...`

**Comment by domcleal** [comment]
Thanks, tests well.

**Comment by isratrade** [comment]
@tstrachota, This looks fine to me although I think it would be more readable if `parse_string_value` did not set the value but rather just returned the parsed value. I think it's fine to merge without refactoring as foreman-config uses `parse_string_value`

**Comment by domcleal** [comment]
Merged as 225b381 for Foreman 1.6.0, thanks @tstrachota.
